libr/main/rasign2.c

```diff
@@ -0,0 +1,119 @@
+#include <stdio.h>
+#include <string.h>
+#include <r_getopt.h>
+#include <r_main.h>
+#include <r_sign.h>
+#include <r_core.h>
+
+#include "r_userconf.h"
+
+static int rasign_show_help() {
+	printf ("Usage: rasign2 [options] [file]\n"
+		" -a [-a]          add extra 'a' to analysis command\n"
+		" -o sigs.sdb      add signatures to file, create if it does not exist\n"
+		" -r               show output in radare commands\n"
+		"Examples:\n"
+		"  rasign2 -o libc.sdb libc.so.6\n");
+	return 0;
+}
+
+static RCore *opencore(const char *fname) {
+	const ut64 baddr = UT64_MAX;
+	RCore *c = r_core_new ();
+	if (!c) {
+		eprintf ("Count not get core\n");
+		return NULL;
+	}
+	r_core_loadlibs (c, R_CORE_LOADLIBS_ALL, NULL);
+	r_config_set_i (c->config, "scr.interactive", false);
+	if (fname) {
+#if __WINDOWS__
+		fname = r_acp_to_utf8 (fname);
```

Comment:
this will be leaking because the function returns char * and fname is const char *

Reply:
I was concerned about this too but was unsure what to do. This code is taken from `radiff2` [here](https://github.com/radareorg/radare2/blob/master/libr/main/radiff2.c#L76).

How do you suggest I resolve this? Add another `#if __WINDOWS__` macro to `free(fname)` after it is used in `r_core_file_open`? Should I change the `const` type?

Is there an OS agnostic function I could use so the string is always allocated with the correct character width, then it can always be free'd. The macro could then be removed.

Reply:
nope, if the string is in the heap and must be freed it should be done like that. you can have two variables, one const and the other non-const, and use one to determine if yo have to free it or not